applications/MappingApplication/custom_utilities/interface_object.h

```diff
@@ -100,117 +99,136 @@ namespace Kratos
-              m_min_distance_neighbor = distance;
-              m_neighbor_rank = rank;
-          }
+      void ProcessSearchResult(const double Distance, const int PairingStatus, const int Rank) {
+          if (mPairingStatus < PairingStatus || (mPairingStatus == PairingStatus 
+                                                 && Distance < mMinDistanceNeighbor)) {
+              mPairingStatus = PairingStatus;
+              mMinDistanceNeighbor = Distance;
+              mNeighborRank = Rank;
+          }   
       }
 
-      bool NeighborFound() {
-          return m_neighbor_found;
+      int GetPairingStatus() {
+          return mPairingStatus;
       }
 
-      // double MinDistance() {
-      //     return m_min_distance_neighbor;
-      // }
+      bool NeighborOrApproximationFound() {
+          if (mPairingStatus >= this->GetIndexApproximation())  {
+              return true;
+          } else {
+              return false;
+          }
+      }
 
-      bool HasNeighborInPartition(const int partition_index) {
+      bool HasNeighborOrApproximationInPartition(const int PartitionIndex) {
           bool return_value = false;
-          if (m_neighbor_found) {
-              if (m_neighbor_rank == partition_index)
+          if (mPairingStatus >= this->GetIndexApproximation()) {
+              if (mNeighborRank == PartitionIndex)
                   return_value = true;
           }
           return return_value;
       }
 
-      void SetIsBeingSent() {
-          m_is_being_sent = true;
+      int GetIndexNoNeighbor() {
+          return 0;
       }
 
-      bool GetIsBeingSent() {
-          return m_is_being_sent;
+      int GetIndexApproximation() {
+          return 1;
+      }
+
+      int GetIndexNeighborFound() {
+          return 2;
       }
 
       int GetNeighborRank() {
-          return m_neighbor_rank;
+          return mNeighborRank;
       }
 
-      virtual int GetObjectId() {
-          KRATOS_ERROR << "MappingApplication; InterfaceObject; \"GetObjectId\" "
-                       << "of the base class called!" << std::endl;
-          return -1;
+      void SetIsBeingSent() {
+          mIsBeingSent = true;
       }
 
-      virtual void PrintMatchInfo() {
-          KRATOS_ERROR << "MappingApplication; InterfaceObject; \"PrintMatchInfo\" "
-                       << "of the base class called!" << std::endl;
+      bool GetIsBeingSent() {
+          return mIsBeingSent;
       }
 
-      virtual bool EvaluateResult(array_1d<double, 3> global_coords, double& min_distance,
-                                  double distance, array_1d<double,2>& local_coords,
-                                  std::vector<double>& shape_function_values) {
-          KRATOS_ERROR << "MappingApplication; InterfaceObject; \"EvaluateResult\" "
-                       << "of the base class called!" << std::endl;
+      virtual bool EvaluateResult(const array_1d<double, 3>& rGlobalCoords, 
```

Comment:
I need instances of the InterfaceObject, thats why I can't make it pure virtual
Furthermore some functions are needed (and hence overridden in the child class) for the InterfaceNode, some for the InterfaceCondition. I intended these Errors to be safety measures in case for some reason the wrong class is initialized.